torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1389,13 +1403,96 @@ void ONNXShapeTypeInference(
 
 } // namespace
 
+std::pair<bool, bool> IsShapeReliable(Node* n) {
+  auto reliable = true;
+  auto complete = true;
+  for (auto input : n->inputs()) {
+    if (ConstantValueMap::HasTypeReliable(input->debugName())) {
+      auto reliable_value =
+          ConstantValueMap::GetTypeReliable(input->debugName()).value();
+      reliable = reliable && reliable_value;
+    } else {
+      reliable = false;
+    }
+    if (auto pt = input->type()->cast<TensorType>()) {
+      if (!pt->sizes().isComplete()) {
+        complete = false;
+      }
+    }
+  }
+  return std::make_pair(reliable, complete);
+}
+
+// There is no need to put onnx type here, but we need this
+// for some legacy tests when onnx_shape_inference=False.
+static std::unordered_set<std::string> nodeTypeReliableForTracer = {
```

Comment:
nit: use `std::unordered_set<::c10::Symbol>`, and set values like `prim::ListConstruct` etc?

Reply:
Tried that, but the contrib op "com.microsoft" does not have such `::c10::Symbol`.